### ISSUE TITLE
Unable to redeploy solution

### ISSUE THREAD

**Issue by SammyC80** [microsoft/startstopv2-deployments#45]
Hello,

First of all thank you for this solution, it's great, however I needed to remove it and redeploy it and the deployment fails.  I have tried redeploying this on 2 tenants now (different subscriptions completely) and get the same issue.

I get bad requests on 3 of the functions as follows:

{
    "status": "Failed",
    "error": {
        "code": "BadRequest",
        "message": "Scope can not be updated"
    }
}

After this I deleted the entire resource group I was deploying to and tried again to a new resource group, it got further but still failed, this time with:

    "status": "Failed",
    "error": {
        "code": "RoleAssignmentUpdateNotPermitted",
        "message": "Tenant ID, application ID, principal ID, and scope are not allowed to be updated."

Although this time all resources seemed to be deployed, when attempting to start a VM it failed with:

{
  "Message": "An error has occurred.",
  "ExceptionMessage": "The client 'xxx' with object id 'xxx' does not have authorization to perform action 'Microsoft.Resources/subscriptions/resources/read' over scope '/subscriptions/xxx' or the scope is invalid. If access was recently granted, please refresh your credentials.",
  "ExceptionType": "Microsoft.Rest.Azure.CloudException",

Please help, I am not doing anything differently to when I initially deployed it to either tenant, it just isn't redeploying, is there some role or something I need to remove prior to redeploying?

Many thanks in advance!

**Comment by svarga-msft** [comment]
When you removed the original deployment did you delete the managed identity associated with it? You might need to go into your subscription access control and find the managed identity and delete it.

From the error it sounds like the managed identity used by start stop is not getting deployed correctly. Is it possible that the subscription permissions have been updated recently? 

You should see your function apps managed identity under the subscriptions access control like this:
![image](https://user-images.githubusercontent.com/68304251/149583473-de6a60df-d9a9-4c65-9c35-9d4da5cd5457.png)

Let me know if you see it and we can go from there.

**Comment by SammyC80** [comment]
Hi there,

I didn't delete the managed identity as I didn't see that needed to be done
in the documentation (
https://docs.microsoft.com/en-us/azure/azure-functions/start-stop-vms/remove).
I looked where you suggested and there appeared to be an orphaned identity
(screenshot below), once this was removed I managed to redeploy the
solution to my own test tenant and then doing the same thing on my
company's production tenant.  Thank you very much for your assistance
it's much appreciated, might be worth adding this requirement into the
otherwise excellent documentation regarding redeployment.

[image: image.png]
